src/app/student/map/_components/MainView.tsx

```diff
@@ -0,0 +1,134 @@
+"use client"
+
+import { MapComponent } from "@/app/student/map/_components/MapComponent"
+import Sidebar from "@/app/student/map/_components/Sidebar"
+import EditorMapComponent from "@/app/student/map/editor/EditorMapComponent"
+import { Exhibitor } from "@/components/shared/hooks/api/useExhibitors"
+import { Button } from "@/components/ui/button"
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue
+} from "@/components/ui/select"
+import { useSearchParams } from "next/navigation"
+import { useState } from "react"
+import { Booth, BoothID, BoothMap } from "../lib/booths"
+import { defaultLocation, LocationId, locations } from "../lib/locations"
+
+export default function MainView({
+  boothsByLocation,
+  boothsById,
+  exhibitorsById
+}: {
+  boothsByLocation: Map<LocationId, BoothMap>
+  boothsById: BoothMap
+  exhibitorsById: Map<number, Exhibitor>
+}) {
+  // url: /student/map?location=[nymble/1|nymble/2|nymble/3|library]&lat=[number]&lng=[number]&zoom=[number]
+  // if location is not provided or is invalid, default to nymble/1
+  // if lat, lng or zoom is not provided, default to location center
+  const searchParams = useSearchParams()
+
+  const locationString = searchParams.get("location") ?? "nymble/1"
+  const locationIdString = locations.some(loc => loc.id === locationString)
+    ? locationString
+    : defaultLocation.id
+  const [locationId, setLocationId] = useState<LocationId>(
+    locationIdString as LocationId
+  )
```

Comment:
Changed the url parameter to "floor" instead of "location" but ended up keeping location and locationId in the code. Agree that this terminology is somewhat confusing but it's already used in so many places in the code that it would be a bit difficult to change. Also not sure what to change it to, I feel like floor would make sense if we only had one building but the library isn't really a floor in my head